File: ExercicioComparacaoLinguagens/atividade_encapsulamento/java/encapsulamento/src/dtos/Profissao.java
package dtos;

class Profissao {
	// codigo acessivel apenas para o pacote dtos
	protected String nome;
	protected float salario;
	protected String nCarteiraTrabalho;

	public Profissao(String nome, float salario, String nCarteiraTrabalho) {
		super();
		this.nome = nome;
		this.salario = salario;
		this.nCarteiraTrabalho = nCarteiraTrabalho;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public float getSalario() {
		return salario;
	}

	public void setSalario(float salario) {
		this.salario = salario;
	}

	public String getnCarteiraTrabalho() {
		return nCarteiraTrabalho;
	}

	public void setnCarteiraTrabalho(String nCarteiraTrabalho) {
		this.nCarteiraTrabalho = nCarteiraTrabalho;
	}

}

File: ExercicioComparacaoLinguagens/atividade_encapsulamento/php/dtos/Profissao.php
<?php
class Profissao {
    protected $nome;
    protected $salario;
    protected $nCarteiraTrabalho;

    public function __construct($nome, $salario, $nCarteiraTrabalho) {
        $this->nome = $nome;
        $this->salario = $salario;
        $this->nCarteiraTrabalho = $nCarteiraTrabalho;
    }

    // Getters e Setters
    public function getNome() {
        return $this->nome;
    }

    public function setNome($nome) {
        $this->nome = $nome;
    }

    public function getSalario() {
        return $this->salario;
    }

    public function setSalario($salario) {
        $this->salario = $salario;
    }

    public function getnCarteiraTrabalho() {
        return $this->nCarteiraTrabalho;
    }

    public function setnCarteiraTrabalho($nCarteiraTrabalho) {
        $this->nCarteiraTrabalho = $nCarteiraTrabalho;
    }

    // Método __toString para exibir os detalhes da profissão
    public function __toString() {
        return "Profissao [nome=" . $this->nome . ", salario=" . $this->salario . ", nCarteiraTrabalho=" . $this->nCarteiraTrabalho . "]";
    }
}
?>
